### ISSUE TITLE
I made a dashboard

### ISSUE THREAD

**Issue by jeffkaufman** [jeffkaufman/nomic#64]
Want to see what PRs are ready to merge?  Want to see who you need to bug?  I made a dashboard:

https://www.jefftk.com/nomic

It's pretty slow because it makes all the requests serially, but if we like it I can rewrite it to be much faster.

**Comment by tnelling** [comment]
Could definitely use some visual work, but it's absolutely useful as-is. The speed doesn't seem like a huge deal, though obviously it would be nice to make it faster.

Does it make sense to incorporate this into the game itself?

**Comment by jeffkaufman** [comment]
Added views for each player.

For example: https://www.jefftk.com/nomic/guoruibiao (Hey @guoruibiao are you still playing?)

**Comment by tnelling** [comment]
That reminds me... one thing we haven't discussed is whether we want to collectively vote to remove inactive players. Relatedly, another thing this dashboard could do is show timestamps (or some other way of indicating how long someone has let a PR sit, and in particular whether they have a stale review vs never reviewed, that kind of thing).

But I don't want to demand a bunch of work from you, @jeffkaufman, which is why I was wondering about integrating the dashboard into the game, or at least making it accessible for players to contribute to.

**Comment by pavellishin** [comment]
> one thing we haven't discussed is whether we want to collectively vote to remove inactive players.

I don't think we *have* to, right? We've got enough active players that we can eventually push PRs through. 

I'm also liking my interpretation of your suggestion about which votes should be counted and when; I think if we implement something like what we talked about in #62, inactive players could be effectively "irrelevant".

**Comment by tnelling** [comment]
True enough, there are multiple ways to tackle this problem. The short-term advantage of removing inactive players is that it becomes easier for us to start pushing through more of the rules that make inactivity irrelevant long-term.

**Comment by jeffkaufman** [comment]
One thing my dashboard doesn't show is, how long has it been since a player contributed in any way at all.  Which isn't something that's easy to get, since it only pulls reviews on open PRs.

**Comment by jeffkaufman** [comment]
I think we should remove inactive players manually, and if they actually want to keep playing they can veto the PR

**Comment by jeffkaufman** [comment]
Made the dashboard a bit prettier

**Comment by jeffkaufman** [comment]
Made the dashboard be much faster: https://www.jefftk.com/nomic

**Comment by tnelling** [comment]
That top section is a nice distillation of the key information. The only thing I really want to be able to see at a glance now that I can't is blocked PRs (so I could verify that my TBDs are waiting on other PRs).

**Comment by jeffkaufman** [comment]
@tnelling PRs that are labeled "blocked" are now no longer bold. (And PRs have a list of labels in the full section below)

Though we might want a new label, since "blocked" is kind of overloaded.  Like, https://github.com/jeffkaufman/nomic/pull/35 is unblocked because @pavellishin needs to rework it after a dependent PR merged. Maybe "review"?

**Comment by jeffkaufman** [comment]
(going to sleep now...)

**Comment by tnelling** [comment]
Yeah, that was more of a "this is something we should think about" comment than a request for immediate changes.

**Comment by jeffkaufman** [comment]
I've made a `reviewme` label and applied it to all the PRs that are currently ready for review: [list](https://github.com/jeffkaufman/nomic/pulls?q=is%3Apr+is%3Aopen+label%3Areviewme).

The top section now groups all PRs without the `reviewme` label into "not ready".

The dashboard should now tell people what they need for figuring out what to review.

**Comment by jeffkaufman** [comment]
Here's what it looks like now:

<img width="557" alt="screen shot 2019-01-18 at 11 10 07 am" src="https://user-images.githubusercontent.com/262566/51398471-acdbf100-1b11-11e9-8775-f83e638c57a4.png">

**Comment by jeffkaufman** [comment]
@TheJhyde @dchudz @guoruibiao @vesche if you weren't keeping up because it was too annoying to figure out what needed doing, now would be a great time to come back!  Just have a look at https://www.jefftk.com/nomic and see the PRs next to your name under "needs review".

**Comment by tnelling** [comment]
We might also want a label that indicates that the PR author should update it to handle merge conflicts or dependencies, and have that integrated into the dashboard so authors can check there for that.

**Comment by vesche** [comment]
@jeffkaufman The dashboard is great, I'm caught up now!

**Comment by jeffkaufman** [comment]
@vesche it works!

**Comment by jeffkaufman** [comment]
Tweaks:

* I removed the not-ready ones from the dashboard, since loading more PRs slows the dashboard down and these ones don't need looking at.

* Made the table a bit more compact by removing extra horizontal spacing.

* For ones that need re-review, made them green-bold if you already approved them before, and red-bold if you haven't.

One thing this doesn't capture is that right now if you reject a PR, then approve it, then the PR changes, your earlier rejection is still active.  I think we should fix that in `validate.py` and not in the dashboard since it's not what people expect.

**Comment by pavellishin** [comment]
Did the per-user view go away?

**Comment by jeffkaufman** [comment]
@pavellishin yes, sorry!  When I rewrote it in JS to make it faster I switched to having a grid view at the top which has the same information that was in the per-user view.

**Comment by jeffkaufman** [comment]
I added a section to the dashboard that shows the per-pr view, similar to the earlier per-user view:

<img width="338" alt="screen shot 2019-01-22 at 2 00 03 pm" src="https://user-images.githubusercontent.com/262566/51558704-26474c80-1e4e-11e9-8866-4088ef1800a8.png">

**Comment by tnelling** [comment]
The fact that it says ready for #67, but #67 isn't actually ready, is obviously misleading. Not sure what you can easily do about that since it's something specific to player removal PRs, though.

**Comment by jeffkaufman** [comment]
Yeah, this is an awkward aspect.  This is more or less predicting whether validate on master will pass.
